lib/CompilerImpl/ClangLibCompiler.cpp

```diff
@@ -282,32 +285,35 @@ std::string ClangLibCompiler::runOptimizer(const std::string &src_IR,
   llvm::Triple moduleTriple(module->getTargetTriple());
   std::string optCPUStr, optFeaturesStr;
   llvm::TargetMachine* optTMachine = nullptr;
-  const llvm::TargetOptions Options = InitTargetOptionsFromCodeGenFlags();
+  const llvm::TargetOptions Options = llvm::codegen::InitTargetOptionsFromCodeGenFlags(moduleTriple);
                                       // llvm static helper function
   std::string lookupError;
   if (moduleTriple.getArch()) {
-    optCPUStr = getCPUStr();            // llvm static helper function
-    optFeaturesStr = getFeaturesStr();  // llvm static helper function
+    optCPUStr = llvm::codegen::getCPUStr();            // llvm static helper function
+    optFeaturesStr = llvm::codegen::getFeaturesStr();  // llvm static helper function
   }
-  const llvm::Target *TheTarget = TargetRegistry::lookupTarget(MArch,
+
+  const llvm::Target *TheTarget = llvm::TargetRegistry::lookupTarget(llvm::codegen::getMArch(),
                                                                moduleTriple,
                                                                lookupError);
   // Some modules don't specify a triple, and this is okay.
   if (!TheTarget) {
+
     optTMachine = nullptr;
   } else {
     optTMachine = TheTarget->createTargetMachine(moduleTriple.getTriple(),
                                                  optCPUStr,
                                                  optFeaturesStr,
                                                  Options,
-                                                 getRelocModel(),
-                                                 getCodeModel(),
+                                                 llvm::codegen::getRelocModel(),
+                                                 llvm::CodeModel::Small,// llvm::codegen::getCodeModel returns zero (casted to Tiny), which is wrong! Going with small which is the default model for majority of supported targets
```

Comment:
does this happen on all versions of llvm?

Reply:
I'll test it and I will update you about it :+1:

Reply:
I get the commented behaviour on llvm 13,14 and 15.
`LLVM ERROR: Target does not support the tiny CodeModel`

Reply:
Can you try with `getExplicitCodeModel()`, which returns an optional, and fallback to Small only if not available?